packages/components/scripts/wrapper-generator/UXPinReactWrapperGenerator.ts

```diff
@@ -440,6 +442,39 @@ export class UXPinReactWrapperGenerator extends ReactWrapperGenerator {
     return [...componentPresetFiles, configFile];
   }
 
+  // Component declaration can be preceded by JSDoc comments
+  // to customize the behavior in UXPin Editor or Preview (E.g.: render in a React Portal)
+  // https://www.uxpin.com/docs/merge/adjusting-components/
+  private insertComponentAnnotation(cleanedComponent: string, component: TagName): string {
+    const comments = this.getAllComponentComments(component);
+    if (!comments.length) return cleanedComponent;
+
+    const annotations = `/**
+${comments.join(`\n`)}
+*/
+`;
+    return annotations + cleanedComponent;
+  }
+
+  private getAllComponentComments(component: TagName): string[] {
+    const comments: string[] = [];
+
+    if (this.shouldRenderInReactPortal(component)) {
+      comments.push('* @uxpinuseportal');
+    }
+    return comments;
+  }
```

Comment:
Now only the `@uxpinportal` feature is supported but my assumption was that we could support more properties from this list:

```
@uxpincomponent
@uxpinnamespace
@uxpinwrappers
@uxpindocurl
@uxpinuseportal
```
 
Of course the implementation could be even simpler if we consider that we only care about `@uxpinuseportal` 😅